domain_utils/domain_utils.py

```diff
@@ -48,88 +30,181 @@ def is_ip_address(hostname):
+
+    if domain_less_ps != '':
+        domain_parts_to_pop = list(reversed(domain_less_ps.split('.')))
+        while len(domain_parts_to_pop) > 0:
+            domain_parts = list(reversed(domain_parts_to_pop)) + [ext.suffix]
+            subparts.append('.'.join(domain_parts))
+            domain_parts_to_pop.pop()
+
     if include_ps:
-        try:
-            subparts.append(ps_plus_1[ps_plus_1.index('.')+1:])
-        except Exception:
-            pass
+        subparts.append(ext.suffix)
+
     return subparts
 
 
-def get_stripped_url(url, scheme=False):
-    """Returns a url stripped to (scheme)?+hostname+path"""
+def get_stripped_url(url, scheme=False, drop_non_http=False, use_netloc=True):
+    """
+    Returns a url stripped to just the beginning and end, or more formally,
+    ``(scheme)?+(netloc|hostname)+(path)?``.
+    For example ``https://my.domain.net/a/path/to/a/file.html#anchor?a=1``
+    becomes ``my.domain.net/a/path/to/a/file.html``
+    URL parsing is done using std lib
+    `urllib.parse.urlparse
+    <https://docs.python.org/3.8/library/urllib.parse.html>`_.
+    Empty scheme e.g. ``my.domain.cloudfront.net``
+    are assumed to be http schemes.
+    If a URL has a port but no scheme, urlparse determines the scheme to
+    be the hostname and we do not handle this special case. In this case,
+    the url will be treated as a non-http scheme and the return value will
+    be determined by the ``drop_non_http`` setting.
+    :param url: URL to be parsed
+    :type url: str
+    :param scheme: If ``True``, scheme will be prepended in
+        returned result. Defaults is ``False``.
+    :type scheme: bool, optional
+    :param drop_non_http: Action to take if scheme is not
+        ``http`` or ``https`` e.g. ``file:`` or ``about:blank``.
+        If ``True``, the result for non http urls will be an empty string
+        If ``False``, the result for non http urls will be the original url,
+        not further processed e.g. ``about:blank`` -> ``about:blank`` even
+        if ``scheme=False``. The result for http urls will be the stripped
+        url with or without the scheme as per scheme param.
+        Default is ``False``.
+    :type drop_non_http: bool, optional
+    :param use_netloc: If ``True`` urlparse's netloc will be used.
+        If ``False`` urlparse's host will be returned. Using netloc means
+        that a port is included, for example, if it was in the path.
+        Default is ``True``.
+    :type use_netloc: bool, optional
+    :return: Returns a url stripped to (scheme)?+(netloc|hostname)+(path)?.
+        Returns empty string if appropriate.
+    :rtype: str
+    """
     purl = urlparse(url)
-    surl = ''
-    if scheme:
-        surl += purl.scheme + '://'
-    try:
-        surl += purl.hostname + purl.path
-    except TypeError:
-        surl += purl.hostname
-    return surl
+    _scheme = purl.scheme
+
+    # Handle non http schemes
+    if _scheme not in ['http', 'https', '']:
```

Comment:
We should add support for `ws` and `wss`. (https://developer.mozilla.org/en-US/docs/Web/API/WebSockets_API/Writing_WebSocket_client_applications)

It would be helpful to be able to retrieve eTLD+1 from these as well.